src/mono/mono/mini/exceptions-x86.c

```diff
@@ -272,6 +298,11 @@ void win32_seh_set_handler(int type, MonoW32ExceptionHandler handler)
 	case SIGSEGV:
 		segv_handler = handler;
 		break;
+	case SIGTERM:
+		term_handler = handler;
+		if (!SetConsoleCtrlHandler(mono_win_ctrl_handler, TRUE))
```

Comment:
This adds a new handler, so if win32_seh_set_handler was previously called to install a different term_handler, shouldn't we uninstall the old one first?

Reply:
actually, i misread, but i think this would add a copy of our handler each time you set a term handler, so we might end up with 3 copies of the same one installed unless the kernel removes duplicates. the docs don't say whether it does

if an application calls AllocConsole or AttachConsole this handler will get uninstalled, but that's probably fine to consider as a 'don't do that' problem